### ISSUE TITLE
MatchingDeclarationName has false positives

### ISSUE THREAD

**Issue by rock3r** [arturbosch/detekt#686]
In this case (filename `Whatever.kt`):

```kotlin
fun a()

private class Something
```

The MatchingDeclarationName rule flags the file — I suppose saying that it should be called `Something.kt` as it's the only class in there — but it shouldn't really as this class contains only a publicly visible top level function and that private class is only an implementation detail of the function. One such file for which this is reproducible is [this](https://github.com/squanchy-dev/squanchy-android/blob/develop/app/src/main/java/net/squanchy/tweets/service/TweetModelConverter.kt).

**Comment by rock3r** [comment]
Aaaand here's a PR to fix it.

**Comment by arturbosch** [comment]
Thanks for the fix!

**Comment by rock3r** [comment]
No worries
